### ISSUE TITLE
not able to make KAT-2.0.7

### ISSUE THREAD

**Issue by zhiliangc** [TGAC/KAT#27]
After a whole bunch of warnings, my ```make``` stopped with
```
Making all in tests
make[2]: Entering directory `/home/zxc561/KAT/tests'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/home/zxc561/KAT/tests'
make[2]: Entering directory `/home/zxc561/KAT'
make[2]: *** No rule to make target `deps/seqan-library-2.0.0.tar.gz', needed by `all-am'.  Stop.
make[2]: Leaving directory `/home/zxc561/KAT'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/zxc561/KAT'
make: *** [all] Error 2
```
I configured with
```
./configure --prefix=$PWD --with-boost-libdir=/apps/boost/1.58.0/lib --with-boost=/apps/boost/1.58.0
```
Was using ```cmake/2.8.12.1``` initially so I thought it was about  ```cmake``` so I then changed to ```cmake/3.0.1``` and the same problem was still there.

I've added the following packages to my PATH, as well as ```autoconfig 2.69 ``` and ```automake/1.15```
```
 1) gnuplot/5.0.1   2) boost/1.58      3) gcc/5.2.0       4) cmake/3.0.1
```

**Comment by zhiliangc** [comment]
With a cleaned configuration and adding ```libtool/2.4.6``` I can now ```make```, but ```make install``` still gives me error.

```
 make install
Making install in deps/jellyfish-2.2.0
make[1]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make  install-recursive
make[2]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
Making install in .
make[3]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[4]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
 /bin/mkdir -p '/home/zxc561/KAT/lib'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libjellyfish-2.0.la '/home/zxc561/KAT/lib'
libtool: install: /usr/bin/install -c .libs/libjellyfish-2.0.so.2.0.0 /home/zxc561/KAT/lib/libjellyfish-2.0.so.2.0.0
libtool: install: (cd /home/zxc561/KAT/lib && { ln -s -f libjellyfish-2.0.so.2.0.0 libjellyfish-2.0.so.2 || { rm -f libjellyfish-2.0.so.2 && ln -s libjellyfish-2.0.so.2.0.0 libjellyfish-2.0.so.2; }; })
libtool: install: (cd /home/zxc561/KAT/lib && { ln -s -f libjellyfish-2.0.so.2.0.0 libjellyfish-2.0.so || { rm -f libjellyfish-2.0.so && ln -s libjellyfish-2.0.so.2.0.0 libjellyfish-2.0.so; }; })
libtool: install: /usr/bin/install -c .libs/libjellyfish-2.0.lai /home/zxc561/KAT/lib/libjellyfish-2.0.la
libtool: install: /usr/bin/install -c .libs/libjellyfish-2.0.a /home/zxc561/KAT/lib/libjellyfish-2.0.a
libtool: install: chmod 644 /home/zxc561/KAT/lib/libjellyfish-2.0.a
libtool: install: ranlib /home/zxc561/KAT/lib/libjellyfish-2.0.a
libtool: finish: PATH="/apps/libtool/2.4.6/bin:/apps/cmake/3.0.1/bin:/apps/autoconfig/2.69/bin:/apps/automake/1.15/bin:/apps/gcc/5.2.0/bin:/apps/gnuplot/5.0.1/bin:/home/zxc561/autoconf/bin:/apps/automake/1.15/bin:/opt/pbs/default/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/X11R6/bin:/usr/games:/usr/lib/mit/bin:/usr/lib/mit/sbin:/opt/sgi/sgimc/bin:/apps/sga-master/bin:/apps/python/2.7.3-2/bin:/sbin" ldconfig -n /home/zxc561/KAT/lib
----------------------------------------------------------------------
Libraries have been installed in:
   /home/zxc561/KAT/lib

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the '-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the 'LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the 'LD_RUN_PATH' environment variable
     during linking
   - use the '-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to '/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------
 /bin/mkdir -p '/home/zxc561/KAT/bin'
  /bin/sh ./libtool   --mode=install /usr/bin/install -c bin/jellyfish '/home/zxc561/KAT/bin'
libtool: install: /usr/bin/install -c bin/.libs/jellyfish /home/zxc561/KAT/bin/jellyfish
 /bin/mkdir -p '/home/zxc561/KAT/include/jellyfish-2.2.0/jellyfish'
 /usr/bin/install -c -m 644 include/jellyfish/allocators_mmap.hpp include/jellyfish/backtrace.hpp include/jellyfish/atomic_gcc.hpp include/jellyfish/large_hash_array.hpp include/jellyfish/err.hpp include/jellyfish/misc.hpp include/jellyfish/offsets_key_value.hpp include/jellyfish/int128.hpp include/jellyfish/rectangular_binary_matrix.hpp include/jellyfish/mer_dna.hpp include/jellyfish/storage.hpp include/jellyfish/simple_circular_buffer.hpp include/jellyfish/circular_buffer.hpp include/jellyfish/atomic_field.hpp include/jellyfish/compare_and_swap.hpp include/jellyfish/divisor.hpp include/jellyfish/large_hash_iterator.hpp include/jellyfish/jellyfish.hpp include/jellyfish/thread_exec.hpp include/jellyfish/stream_iterator.hpp include/jellyfish/mer_overlap_sequence_parser.hpp include/jellyfish/whole_sequence_parser.hpp include/jellyfish/binary_dumper.hpp include/jellyfish/sorted_dumper.hpp include/jellyfish/text_dumper.hpp include/jellyfish/dumper.hpp include/jellyfish/time.hpp include/jellyfish/mer_heap.hpp include/jellyfish/token_ring.hpp include/jellyfish/locks_pthread.hpp include/jellyfish/file_header.hpp include/jellyfish/generic_file_header.hpp include/jellyfish/json.h include/jellyfish/hash_counter.hpp include/jellyfish/mapped_file.hpp include/jellyfish/mer_dna_bloom_counter.hpp include/jellyfish/bloom_common.hpp include/jellyfish/bloom_counter2.hpp include/jellyfish/bloom_filter.hpp include/jellyfish/cooperative_pool.hpp '/home/zxc561/KAT/include/jellyfish-2.2.0/jellyfish'
 /usr/bin/install -c -m 644 include/jellyfish/cooperative_pool2.hpp include/jellyfish/stream_manager.hpp include/jellyfish/generator_manager.hpp include/jellyfish/cpp_array.hpp include/jellyfish/mer_iterator.hpp include/jellyfish/atomic_bits_array.hpp include/jellyfish/stdio_filebuf.hpp include/jellyfish/mer_qual_iterator.hpp '/home/zxc561/KAT/include/jellyfish-2.2.0/jellyfish'
 /bin/mkdir -p '/home/zxc561/KAT/share/man/man1'
 /usr/bin/install -c -m 644 'doc/jellyfish.man' '/home/zxc561/KAT/share/man/man1/jellyfish.1'
 /bin/mkdir -p '/home/zxc561/KAT/lib/pkgconfig'
 /usr/bin/install -c -m 644 jellyfish-2.0.pc '/home/zxc561/KAT/lib/pkgconfig'
make[4]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[3]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[2]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[1]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
Making install in src
make[1]: Entering directory `/home/zxc561/KAT/src'
make[2]: Entering directory `/home/zxc561/KAT/src'
 /bin/mkdir -p '/home/zxc561/KAT/bin'
  /usr/bin/install -c kat '/home/zxc561/KAT/bin'
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/home/zxc561/KAT/src'
make[1]: Leaving directory `/home/zxc561/KAT/src'
Making install in tests
make[1]: Entering directory `/home/zxc561/KAT/tests'
make[2]: Entering directory `/home/zxc561/KAT/tests'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/home/zxc561/KAT/tests'
make[1]: Leaving directory `/home/zxc561/KAT/tests'
make[1]: Entering directory `/home/zxc561/KAT'
make[2]: Entering directory `/home/zxc561/KAT'
 /bin/mkdir -p '/home/zxc561/KAT/bin'
 /usr/bin/install -c scripts/dist_analysis.py '/home/zxc561/KAT/bin'
 /bin/mkdir -p '/home/zxc561/KAT/share/kat'
 /usr/bin/install -c -m 644 deps/seqan-library-2.0.0.tar.gz '/home/zxc561/KAT/share/kat'
make  install-data-hook
make[3]: Entering directory `/home/zxc561/KAT'
cd /deps; \
	cat deps/seqan-library-2.0.0.tar.gz | gunzip |  -H gnu --list > uninstall_manifest.txt; \
	cat deps/seqan-library-2.0.0.tar.gz | gunzip |  -H gnu --no-same-owner --extract; \
	rm --force deps/seqan-library-2.0.0.tar.gz; \
	cat uninstall_manifest.txt | sed --expression='s/^\|$/"/g' | xargs chmod a=rX,u+w
/bin/sh: line 0: cd: /deps: No such file or directory
/bin/sh: line 1: -H: command not found
/bin/sh: line 2: -H: command not found
chmod: missing operand after `a=rX,u+w'
Try `chmod --help' for more information.
make[3]: *** [install-data-hook] Error 123
make[3]: Leaving directory `/home/zxc561/KAT'
make[2]: *** [install-data-am] Error 2
make[2]: Leaving directory `/home/zxc561/KAT'
make[1]: *** [install-am] Error 2
make[1]: Leaving directory `/home/zxc561/KAT'
make: *** [install-recursive] Error 1
```

**Comment by zhiliangc** [comment]
```make check``` shows
```
Making check in deps/jellyfish-2.2.0
make[1]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make  check-recursive
make[2]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
Making check in .
make[3]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make  libgtest.la libgtest_main.la bin/generate_sequence bin/test_all
make[4]: Entering directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
  CXX      unit_tests/gtest/src/libgtest_la-gtest-all.lo
  CXXLD    libgtest.la
  CXX      unit_tests/gtest/src/libgtest_main_la-gtest_main.lo
  CXXLD    libgtest_main.la
  CXX      jellyfish/generate_sequence.o
  CXX      jellyfish/mersenne.o
  CXX      jellyfish/backtrace.o
  CXX      jellyfish/dbg.o
  CXXLD    bin/generate_sequence
  CXX      unit_tests/bin_test_all-test_main.o
In file included from ./unit_tests/gtest/gtest.h:308:0,
                 from unit_tests/test_main.cc:20:
/plocal/current/gcc/5.2.0/include/c++/5.2.0/sstream:300:7: error: ‘struct std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::__xfer_bufptrs’ redeclared with different access
       struct __xfer_bufptrs
       ^
make[4]: *** [unit_tests/bin_test_all-test_main.o] Error 1
make[4]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[3]: *** [check-am] Error 2
make[3]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[2]: *** [check-recursive] Error 1
make[2]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make[1]: *** [check] Error 2
make[1]: Leaving directory `/home/zxc561/KAT/deps/jellyfish-2.2.0'
make: *** [check-recursive] Error 1
```

**Comment by tseemann** [comment]
This looks like a similar bug I had here: https://github.com/TGAC/KAT/issues/22

**Comment by zhiliangc** [comment]
Thanks @tseemann .
I had a look at my Makefile, seems the tabs were correctly placed.... 

@maplesond Is it possible to have a list of required packages included in the README in the next release? I had to try quite a few of times to get the correct combination of dependencies

**Comment by maplesond** [comment]
I'll take a look at the detail of this today.  I should mention libtool in the readme.  Cmake is unnecessary however.  This is not a cmake project.  I suspect there could be issues with make check, which again I'll have a look at today.

**Comment by maplesond** [comment]
- I'll fix the `make install` issue.  Details described in https://github.com/TGAC/KAT/issues/28 

- I can't reproduce your specific `make check` problem.  On my system it compiles fine.  Possibly this is an issue with the compiler version.  I'll investigate whether this might be due to using GCC V5+.  I should point out that there are further issues with `make check` in that some jellyfish unit tests no longer pass due to modifications I've made to the code.  I'll either remove or modify these tests as appropriate.

- I'll update the README to include requirements for gcc V4.9+, autoconf V2.53+, automake V1.11+,  libtool V2.4.2+, make and boost V1.53+.  Gnuplot is optional.

These changes will go into a V2.0.8 release shortly.  Does this cover all your points?

**Comment by zhiliangc** [comment]
Thanks a lot for working on this.
I'll wait patiently for the next update :)

**Comment by maplesond** [comment]
I think all your issues are fixed in V2.0.8.  I'm closing the case for now, but feel free to reopen if there's anything I've missed.
